src/app/index.ts

```diff
@@ -0,0 +1,196 @@
+/*
+ * Copyright (C) 2018 Red Hat, Inc. and others.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+import yosay = require('yosay');
+import Base = require('yeoman-generator');
+module.exports = class TheiaPlugin extends Base {
+
+    private params!: {
+        author: string
+        version: string
+        license: string
+        pluginName: string
+        pluginType: string
+        githubURL: string
+        extensionPrefix: string
+        example: boolean
+        theiaVersion: string
+        pluginSourcePath: string
+        pluginDistPath: string
+        isFrontend: boolean
+        isBackend: boolean
+    };
+
+    constructor(args: string | string[], options: any) {
+        super(args, options);
+        this.argument('pluginName', {
+            type: String,
+            required: false,
+        });
+
+        this.option('pluginType', {
+            alias: 't',
+            description: 'Type of the plugin [backend, frontend]',
+            type: String,
+        });
+
+        this.option('author', {
+            alias: 'a',
+            description: 'The extension\'s author',
+            type: String
+        });
+        this.option('version', {
+            alias: 'v',
+            description: 'The plugin\'s version',
+            type: String,
+            default: '0.0.1'
+        });
+        this.option('description', {
+            alias: 'd',
+            description: 'The plugin\'s description',
+            type: String,
+        });
+        this.option('license', {
+            alias: 'l',
+            description: 'The plugin\'s license',
+            type: String
+        });
+        this.option('githubURL', {
+            alias: 'u',
+            description: 'The plugin\'s Github URL',
+            type: String
+        });
+
+        this.option('theia-version', {
+            alias: 't',
+            description: 'The version of Theia to use',
+            type: String,
+            default: 'latest'
+        });
+    }
+
+    path() {
+        this.sourceRoot(__dirname + '/../../templates');
+    }
+
+    getName() {
+
+    }
+
+    async doPrompt() {
+
+        if (!(this.options as any).pluginName) {
+            const answers = await this.prompt([{
+                type: 'input',
+                name: 'name',
+                message: "What is the plugin's name",
```

Comment:
It seems question mark is missed at the end of this message.

Reply:
fixed, thanks